src/app/browse-by/browse-by-metadata-page/browse-by-metadata-page.component.ts

```diff
@@ -152,9 +152,7 @@ export class BrowseByMetadataPageComponent implements OnInit, OnDestroy {
           this.browseId = params.id || this.defaultBrowseId;
           this.authority = params.authority;
 
-          if (typeof params.value === 'string'){
-            this.value = params.value.trim();
-          }
+          this.value = +params.value || params.value || '';
```

Comment:
I think reverting this line is still problematic, as the `+params.value` will remove leading zeros.  If you search for "0905" it trims it to "905".  So, we may need to fix this in a different way.

Reply:
Ah, I didn't see #2128. That makes sense. Your suggestion will work. I'll make the change.